app/src/router.js

```diff
@@ -7,29 +7,35 @@ import { setNetwork } from "./scripts/setNetwork"
 Vue.use(router)
 
 export const routeGuard = (store) => async (to, from, next) => {
-  // Set any open modal to false
+  // set any open modal to false
   store.state.session.currrentModalOpen = false
 
-  // Redirect if fullPath begins with a hash (fallback for old pre history mode urls)
+  // fallback for old history mode url redirect if path includes a hash
   if (to.fullPath.includes("#")) {
     const path = to.fullPath.substr(to.fullPath.indexOf("#") + 1)
     next(path)
-    return
   }
+
+  // redirect from notifications to paywall is user is not signed in
   if (to.name === `notifications`) {
     if (store.state.account.userSignedIn) {
       next()
+      // this is needed to prevent accessing notifications from paywall
```

Comment:
interesting! didn't realize this was the fix. but also, this feels hacky. do you know why double clicking would override the guard?

Reply:
did you debug what happens on the 2nd click? I think the bug with the 2nd click was on the other PR though.

Reply:
also, i kind of like this being broken for now because otherwise there is no way to access notifications 😆

Reply:
2nd click? I mean, you are just taking to notifications for some reason, it ignores the `next(/paywall)`

Reply:
> it ignores the next(/paywall)

yes. why?

Reply:
possibly: https://router.vuejs.org/guide/advanced/navigation-guards.html#global-before-guards

```
Make sure that the next function is called exactly once in any given pass through the navigation guard. It can appear more than once, but only if the logical paths have no overlap, otherwise the hook will never be resolved or produce errors. Here is an example of redirecting to user to /login if they are not authenticated:
```

Reply:
let'S fix the double click. try adding a return

> Make sure that the next function is called exactly once

that is what I meant before.

Reply:
Okok. I can try that